modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/logging/CorrelationLoggingTest.java

```diff
@@ -203,71 +209,95 @@ public void testSpecificCorrelationLoggingConfigsTest() throws Exception {
         //Test HTTP
         log.info("Enabling HTTP component correlation logs");
         configureCorrelationLoggingComponent(new String[]{"http"}, true);
-        String logLine;
         InvokeTestAPI();
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***s1: " + logLine);
-            assertTrue(isHTTPLogLine(logLine));
-        }
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***s1: " + logLine);
+                assertTrue(isHTTPLogLine(logLine));
+            }
+        });
+
         log.info("Disabling HTTP component correlation logs");
         configureCorrelationLoggingComponent(new String[] { "http" }, false);
         InvokeTestAPI();
-        while ((logLine = bufferedReader.readLine()) != null) {
-            log.info("***s2: " + logLine);
-            assertTrue(isHTTPLogLine(logLine));
-        }
+        await().atMost(20, TimeUnit.SECONDS).untilAsserted(()-> {
+            String logLine;
+            while ((logLine = bufferedReader.readLine()) != null) {
+                log.info("***s2: " + logLine);
+                assertTrue(isHTTPLogLine(logLine));
```

Comment:
Comment why this is 'true'

Reply:
It was there before as well. however this checks the "HTTP-In" type log message logged by the correlation configuration API

Reply:
Added in https://github.com/wso2/product-apim/pull/13088/commits/76b75f55d3d55e63cac095ea10102e6b155f64eb